integration-tests/tests/src/tests/mixed.ts

```diff
@@ -220,23 +345,882 @@ describe("Mixed", () => {
       expect(value.nullable_list[4]).equals(5, "Should be equal 5");
     });
   });
-  describe("Mixed arrays", () => {
-    openRealmBefore({ schema: [MixedSchema] });
-    it("throws when creating an array of multiple values", function (this: RealmContext) {
-      const objectsBefore = this.realm.objects(MixedSchema.name);
-      expect(objectsBefore.length).equals(0);
-
-      // check if the understandable error message is thrown
-      expect(() => {
+
+  describe("Collection types", () => {
+    openRealmBeforeEach({
+      schema: [
+        MixedSchema,
+        MixedAndEmbeddedSchema,
+        MixedWithDefaultCollectionsSchema,
+        CollectionsOfMixedSchema,
+        EmbeddedObjectSchema,
+        TestObject,
+      ],
+    });
+
+    function expectMatchingFlatList(value: unknown) {
+      expect(value).instanceOf(Realm.List);
+      const list = value as Realm.List<any>;
+      expect(list.length).to.be.greaterThanOrEqual(flatListAllTypes.length);
+
+      let index = 0;
+      for (const item of list) {
+        if (item instanceof Realm.Object) {
+          // @ts-expect-error Property `value` does exist.
+          expect(item.value).equals(unmanagedRealmObject.value);
+        } else if (item instanceof ArrayBuffer) {
+          expectUint8Buffer(item);
+        } else {
+          expect(String(item)).equals(String(flatListAllTypes[index]));
+        }
+        index++;
+      }
+    }
+
+    function expectMatchingFlatDictionary(value: unknown) {
+      expect(value).instanceOf(Realm.Dictionary);
+      const dictionary = value as Realm.Dictionary<any>;
+      expect(Object.keys(dictionary).length).to.be.greaterThanOrEqual(Object.keys(flatDictionaryAllTypes).length);
+
+      for (const key in dictionary) {
+        const value = dictionary[key];
+        if (key === "realmObject") {
+          expect(value).instanceOf(Realm.Object);
+          expect(value.value).equals(unmanagedRealmObject.value);
+        } else if (key === "uint8Buffer") {
+          expectUint8Buffer(value);
+        } else {
+          expect(String(value)).equals(String(flatDictionaryAllTypes[key]));
+        }
+      }
+    }
+
+    function expectUint8Buffer(value: unknown) {
+      expect(value).instanceOf(ArrayBuffer);
+      expect([...new Uint8Array(value as ArrayBuffer)]).eql(uint8Values);
+    }
+
+    describe("Flat collections", () => {
+      describe("CRUD operations", () => {
+        it("can create and access a list with different types (input: JS Array)", function (this: RealmContext) {
```

Comment:
As this sentence gets repeated, would it make sense to put this description into a separate `describe?

```suggestion
        describe("creating and accessing a list with different types", () => {
          it("handles JS Array", function (this: RealmContext) {
```

Reply:
Found a middle ground, I've grouped the CRUD tests into subsuites.